src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -476,10 +476,24 @@ public function __clone()
      */
     public function __toString()
     {
+        $protocol = '';
+        try {
+            $protocol = $this->server->get('SERVER_PROTOCOL');
+        } catch (\InvalidArgumentException $mallformedPath) {
+            trigger_error($mallformedPath->getMessage(), E_USER_ERROR);
```

Comment:
If you have xdebug or the symfony_debug extensions, then you can get the stack trace of fatal error. This is how such issue are easily debugged. No need to keep "long distance relationships" within the code base.